torch/csrc/jit/passes/utils/subgraph_utils.cpp

```diff
@@ -96,28 +103,29 @@ Node* executeSubgraphMergeAndUpdateAliasing(
 // Combine the nodes in two subgraph together. The nodes will end up in
 // `mergeTo`, and `mergeFrom` is destroyed.
 void mergeSubgraph(Node* mergeTo, Node* mergeFrom) {
+  bool merge_from_is_after = mergeFrom->isAfter(mergeTo);
   Node* nodeBeforeMergeFrom = mergeFrom->prev();
   Node* nodeAfterMergeFrom = mergeFrom->next();
 
   unmergeSubgraph(mergeFrom);
 
-  std::vector<Node*> nodes;
-  const auto end_it = nodeBeforeMergeFrom->reverseIterator();
-  auto it = nodeAfterMergeFrom->reverseIterator();
+  graph_node_list_iterator end_it;
+  graph_node_list_iterator it;
+
+  if (merge_from_is_after) {
+    it = nodeBeforeMergeFrom->iterator();
+    end_it = nodeAfterMergeFrom->iterator();
+  } else {
+    end_it = nodeBeforeMergeFrom->reverseIterator();
+    it = nodeAfterMergeFrom->reverseIterator();
+  }
   ++it;
 
-  // defer destroying nodes until after all nodes have been merged,
-  // to make iterators easier to reason about
   std::vector<Node*> merged_nodes;
   while (it != end_it) {
     Node* node = *it;
     ++it;
-    merged_nodes.push_back(node);
-    mergeNodeIntoSubgraph(node, mergeTo, /*destroyNode*/ false);
-  }
-
-  for (Node* n : merged_nodes) {
-    n->destroy();
+    mergeNodeIntoSubgraph(node, mergeTo);
```

Comment:
` /*destroyNode*/ false` do you know why this was an issue before?

Reply:
this could have been removed with the removal of the vmap API PR, I think it existed for that reason